drivers/power/adm1177/adm1177.c

```diff
@@ -0,0 +1,311 @@
+
+	if(ret)
+		return -1;
+
+	*conv_voltage = (data[0] << 4) | (data[2] >> 4);
+
+	*conv_current = (data[1] << 4) | (data[2] & 0xF);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief - Gets the voltage readback from the ADM1177s I2C ADC,if the device
+ *          is initialized right and transmits the requested value.
+ *
+ * @param dev - The device structure.
+ *
+ * @param conv_voltage - The variable through which the voltage is read from the
+ *                       ADC.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int32_t adm1177_voltage_readback(struct adm1177_dev *device,
+				 uint16_t* conv_voltage)
+{
+	int32_t ret;
+
+	uint8_t data[2];
+
+	ret = no_os_i2c_read(device->i2c_desc, data, 2, 1);
+
+	if(ret)
+		return -1;
+
+	*conv_voltage = (data[0] << 4) | (data[1] >> 4);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief - Gets the current readback from the ADM1177s I2C ADC,if the device
+ *          is initialized right and transmits the requested value.
+ *
+ * @param dev - The device structure.
+ *
+ * @param conv_current - The variable through which the current is read from the
+ *                       ADC.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int32_t adm1177_current_readback(struct adm1177_dev *device,
+				 uint16_t* conv_current)
+{
+	int32_t ret;
+
+	uint8_t data[2];
+
+	ret = no_os_i2c_read(device->i2c_desc, data, 2, 1);
+
+	if(ret)
+		return -1;
+
+	*conv_current = (data[0] << 4) | (data[1] >> 4);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief - Converts the raw sample from the Readback into mV.
+ *
+ * @param raw_sample - The raw sample of the Voltage Readback from the ADC.
+ *
+ * @param v_fullscale - 0 if V_FULLSCALE_0 is selected(VRANGE bit is 0).
+ *                      1 if V_FULLSCALE_1 is selected(VRANGE bit is 1).
+ *
+ * @return - Conversion result in mV.
+*******************************************************************************/
+int adm1177_voltage_conversion(uint16_t raw_sample,
+			       int v_fullscale)
+{
+	int voltage = 0;
+
+	if(v_fullscale == 0)
+		voltage = ((V_FULLSCALE_0*1000)/4096) * (int)raw_sample;
+
+	return voltage;
+}
+
+/***************************************************************************//**
+ * @brief - Converts the raw sample from the Readback into uA.
+ *
+ * @param raw_sample - The raw sample of the Current Readback from the ADC.
+ *
+ * @return - Conversion result in uA.
+*******************************************************************************/
+int adm1177_current_conversion(uint16_t raw_sample)
+{
+	int current = 0;
+
+	current = (((1000*105840)/4096)*(int)raw_sample)/R_SENSE;
```

Comment:
- no need to cast to int
- add spaces between arithmetic operators.
- return directly

Reply:
Didn't cast int anymore, added spaces and returned directly.